File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.cpp
	class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        int ans = 0, i = 0, j = 0, n = nums.size(),x = 0,a = 0, b = 0;
        while(j<n){
            a = 0, b = 0;
            while(j<n && (x+(nums[j]&1) <= k)){
                x += (nums[j]&1);
                if(x == k){
                    a++;
                }
                j++;
            }
            while(i<n && x == k){
                if(x==k){
                    b++;
                }
                x -= (nums[i]&1);
                i++;
            }
            ans += a*b;
        }
        return ans;
    }
};

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.py
from collections import defaultdict
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        for index in range(len(nums)):
            if nums[index] % 2 == 0:nums[index] = 0;
            else:nums[index] = 1;
        prefix_map = defaultdict(int)
        current_sum,nice_count = 0,0
        for r in nums:
            current_sum += r
            if current_sum == k:nice_count += 1;
            if (current_sum-k) in prefix_map:
                nice_count += prefix_map[current_sum-k]
            prefix_map[current_sum] += 1
        return nice_count
            
                
            